### PR TITLE
Add a wheel for arm64 macos

### PR DESCRIPTION
**Pull request by glandium**

### CODE REVIEW

**Comment by glandium** [comment]
Ideally, it would be possible to build wheels for python 3.8, but cibuildwheel doesn't allow it (it can't use the python from xcode/command line tools).

**Comment by glandium** [comment]
Note that I tested the arm64 python 3.8 wheel produced from running a reduced workflow on my fork of the repo, and it works. I don't have python 3.9 to test.

File: ci/requirements.macoswheels.in [hunk 1]
```diff
@@ -1 +1 @@
-cibuildwheel
+cibuildwheel==1.9.0
```
Comment (indygreg) [inline ci/requirements.macoswheels.in:1]: We want the `.in` file to not have versions pinned unless there is an incompatibility. Instead, keep this file as-is and run `pip-compile -U --generate-hashes --output-file=ci/requirements.macoswheels.txt ci/requirements.macoswheels.in`. That `-U` tells it to upgrade all packages.

**Review by indygreg — COMMENTED** [review]
I'll have access to an M1 machine in ~10 days. Unless you have objections, I'm going to sit on this PR until I can properly test the wheels.

**Comment by glandium** [comment]
Have you had the chance to look now that you have your M1 according to your tweets? :)

**Review by indygreg — APPROVED** [review]
This looks good to me! Using the "system" Python outside of arm64 builds is a bit wonky. But it shouldn't matter: cibuildwheel should take care of ensuring things behave as expected. I'm not too worried about this doing anything wrong. Although I wouldn't be surprised if I change the logic in the future to only be for arm64 builds. Although hopefully in the future we have arm64 Apple machines in GitHub Actions and this whole mess can be simplified.

**Comment by indygreg** [comment]
I pushed this as 1077372ef8..dae2aab77 but GitHub couldn't figure that out. Thanks for the contribution.

**Comment by glandium** [comment]
Do you have an ETA for next release?

**Comment by indygreg** [comment]
Would you like there to be a release soon?

I wasn't planning to release any time soon since there aren't any major changes queued up. But if there's a business case for releasing, I could do it in the next few days.

**Comment by glandium** [comment]
The arm64 wheels would be useful. A 0.15.x release with them would be nice to have, but if you don't have the bandwidth for that, I can wait.

**Comment by indygreg** [comment]
I've got the 0.15.2 release queued up. Waiting on GitHub Actions to finish processing the wheels. (The wait times on macOS machines can be really bad sometimes.) I should have things published to PyPI within a few hours.

**Comment by indygreg** [comment]
0.15.2 was just published to PyPI.

**Pull request closed without merging**